### DISCUSSION TITLE
Easier dispatchEvent shortcuts mappings

### DISCUSSION THREAD

**Discussion by Bellavene** [gdh1995/vimium-c#858] [Q&A]
Would be perfect if the scripting of dispatchEvent for remapping of physical key's gets a bit more transparent. Right now it is a mess, as I see it.

**Comment by gdh1995** [comment]
Um, it has been simplified a bit. When you want to dispatch a keyboard event, it supports such parameters:
1. `ctrlKey`, `metaKey`, `altKey`, and `shiftKey`: boolean; and if a name exists without a suffix of `=false` then it is true
2. `type`: string; may be `keydown`, `keypress` and `keyup`
- 1. when command count is negative, then `keydown` will be replaced with `keyup`
4. `key`: a string which has up to 3 parts joined by a non-word character.
- 1. the 1st part is `event.key`; the 2nd part is `event.keyCode`; and the 3rd is `event.code`
- 2. if there's no 2nd part, while the 1st part is a English letter (a-z), a number (0-9), or `Space`, `Comma`, `Slash`, `Minus`, then Vimium C will auto calculate a suitable `.key`, `.keyCode` and `.code`

Then you just need to map a key with a command sequence (tree) to dispatch both `keydown` and `keyup`, just like https://github.com/gdh1995/vimium-c/wiki/Auto-run-a-tree-of-commands#example , and, in a latest version of Vimium C the `%40` can be replaced with `/` .
